### PR TITLE
sql: add SHOW LOGICAL REPLICATION JOBS syntax

### PR DESCRIPTION
**Pull request by azhu-crl**
This PR creates a noop `SHOW LOGICAL REPLICATION JOBS` SQL syntax. Currently the command does not return any rows since the LDR observability logic will be implemented in a followup PR.

```
demo@127.0.0.1:26257/demoapp/movr> SHOW LOGICAL REPLICATION JOBS;
SHOW LOGICAL REPLICATION JOBS 0

Time: 1ms total (execution 1ms / network 0ms)

demo@127.0.0.1:26257/demoapp/movr> SHOW LOGICAL REPLICATION JOBS WITH DETAILS;
SHOW LOGICAL REPLICATION JOBS 0

Time: 1ms total (execution 0ms / network 0ms)
```

Epic: CRDB-40869
Release note: None

### CODE REVIEW

**Comment by msbutler** [comment]
Off to a good start! In this first PR, it would be great to create a new test like the following which passes
https://github.com/msbutler/cockroach/blob/butler-fix-mixed-version-complete/pkg/sql/parser/testdata/create_logical_replication#L2

File: pkg/sql/sem/tree/show.go [hunk 1]
```diff
@@ -1255,6 +1255,30 @@ func (node *ShowTenant) Format(ctx *FmtCtx) {
...
+	ShowLogicalReplicationJobsOptions
+}
+
+// Format implements the NodeFormatter interface.
+func (node *ShowLogicalReplicationJobs) Format(ctx *FmtCtx) {
+	ctx.WriteString("SHOW LOGICAL REPLICATION JOBS")
+
+	withs := []string{}
+	if node.WithDetails {
+		withs = append(withs, "DETAILS")
```
Comment (msbutler) [inline pkg/sql/sem/tree/show.go:1274]: nit: this could be simplified: could just add `ctx.WriteString(WITH DETAILS)` and remove `withs`

**Review by msbutler — APPROVED** [review]
Nice Job!! One tiny nit: when demonstrating sql output in github, copy and paste as inline code formatting, so it's easier to read/copy/ and paste.
```
plaintext
```

File: pkg/sql/show_logical_replication_jobs.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+var showLogicalReplicationJobsCols = colinfo.ResultColumns{
+	{Name: "job_id", Typ: types.Int},
+	{Name: "targets", Typ: types.StringArray},
+	{Name: "status", Typ: types.String},
+	{Name: "replicated_time", Typ: types.TimestampTZ},
+}
+
+var withDetailsCols = colinfo.ResultColumns{
+	{Name: "replicated_start_time", Typ: types.TimestampTZ},
+	{Name: "retained_time", Typ: types.TimestampTZ},
```
Comment (stevendanna) [inline pkg/sql/show_logical_replication_jobs.go:30]: I don't think we'll have a retained time since we don't have a cutover process.

**Comment by azhu-crl** [comment]
TFTR!

bors r=msbutler

**Pull request merged**